svd-rs/src/array.rs

```diff
@@ -0,0 +1,105 @@
+use super::{DimElement, Name};
+use core::ops::{Deref, DerefMut};
+
+/// A single SVD instance or array of instances
+#[derive(Clone, Debug, PartialEq)]
+pub enum SingleArray<T> {
```

Comment:
I think the name `SingleArray` is a bit confusing. Not sure what alternatives there are though.

Reply:
I'm also not fun on this. I named this `SingleArray` because I thought about changing this enum in future to:
```rust
pub enum SingleArray<T> {
    /// A single instance
    Single(T),
    /// An array of instances
    Array(Array<T>),
}
```
where
```rust
pub struct Array<T> {
    dim: DimElement,
    info: T
}
```
This would let us add some useful methods on `Array`. But this is breaking change, especially in `match` expessions.

Reply:
`MaybeArray`